proto/viam/app/v1/app.proto

```diff
@@ -850,6 +903,27 @@ message UpdateRegistryItemRequest {
 
 message UpdateRegistryItemResponse {}
 
+enum RegistryItemStatus {
+  REGISTRY_ITEM_STATUS_UNSPECIFIED = 0;
+  REGISTRY_ITEM_STATUS_PUBLISHED = 1;
+  REGISTRY_ITEM_STATUS_IN_DEVELOPMENT = 2;
+}
+
+message ListRegistryItemsRequest {
```

Comment:
[opt/pedantic/might be too late to change] Should this be something like `SearchRegistryItemsRequest`.

It seems like this is more than just a basic listing

Reply:
ah agreed that is maybe a better name, but at this point all of our "return multiple items of this type" functions are called "List<item>" (see "ListAuthorizations", "ListLocations", "ListRobots" etc). it's not too late to change but i'm inclined to leave it as is

Reply:
but let me know if you feel strongly -- could be convinced that this is a different enough use case that it warrants "search"

Reply:
Lets stick with List because it matches our conventions. A lot of List APIs have optional filters on them.